zerver/lib/queue.py

```diff
@@ -52,8 +52,14 @@ def _get_parameters(self) -> pika.ConnectionParameters:
         # it doesn't make sense with BlockingConnection
         credentials = pika.PlainCredentials(settings.RABBITMQ_USERNAME,
                                             settings.RABBITMQ_PASSWORD)
+        tcp_options = None
+        if self.rabbitmq_heartbeat == 0:
```

Comment:
I assume it's intentional that is meant to not apply for BlockingConnection; probably worth a comment explaining this.